modules/ui/src/main/scala/lucuma/ui/forms/FormInputEV.scala

```diff
@@ -99,34 +106,87 @@ object FormInputEV {
       props.onValidChange(validated.isValid) >> cb(validated)
     }
 
+    def getInputElement(e: html.Element): CallbackOption[html.Input] = {
+      def childChain(ele: html.Element): Chain[html.Element] =
+        Chain.fromSeq(
+          (0 until ele.childElementCount).map(i => ele.children.item(i).asInstanceOf[html.Element])
+        )
+
+      def findWithClass(ele: html.Element, acc: Chain[html.Element]): Chain[html.Element] = {
+        val newAcc   = if (ele.className.contains("input")) acc :+ ele else acc
+        val children = childChain(ele)
+        if (children.isEmpty) newAcc
+        else newAcc ++ children.flatMap(c => findWithClass(c, Chain.empty))
+      }
+      // val oc = findWithClass(e, Chain.empty).headOption.map(_.firstChild.asInstanceOf[html.Input])
+      CallbackOption.liftOption(
+        findWithClass(e, Chain.empty).headOption.map(_.firstChild.asInstanceOf[html.Input])
+      )
+    }
+
+    def getCursor: CallbackTo[Option[(Int, Int)]] =
+      outerRef.get.flatMap(getInputElement).map(i => (i.selectionStart, i.selectionEnd)).asCallback
+
+    def setCursor(cursor: (Int, Int)): Callback =
+      outerRef.get.flatMap(getInputElement).map(i => i.setSelectionRange(cursor._1, cursor._2))
+
+    def setStateCursor(offset: Int): Callback =
+      getCursor
+        .map(oc => oc.map { case (start, end) => (start + offset, end + offset) })
+        .flatMap(oc => $.setStateL(State.cursor)(oc))
+
+    def clearStateCursor: Callback = $.setStateL(State.cursor)(None)
+
+    def setCursorFromState: Callback =
+      $.state.flatMap(s => s.cursor.map(setCursor).getOrElse(Callback.empty))
```

Comment:
`scalajs-react` has a `.getOrEmpty` method to shorthand this.

Reply:
Nice.